Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -203,4 +302,22 @@ Vector<StyledMarkedText> StyledMarkedText::coalesceAdjacentWithEqualDecorations(
     });
 }
 
+void StyledMarkedText::removeOverlappingForegroundRanges(Vector<StyledMarkedText>& markedTexts)
+{
+    Vector<std::pair<unsigned, unsigned>> ranges;
```

Comment:
can this be a set instead of a vector?

Reply:
I think so! Do I need to do anything special for `ListHashSet` to get it to work with a pair?

Reply:
It could probably just be a HashSet, since this one doesn't need to be ordered, right? And I don't think you'd need to do anything special, assuming our HashSet is smart enough to handle tuples

Reply:
Yes, it doesn't need to be ordered. I'll look into using HashSet now!